internal/keepers/ocr.go

```diff
@@ -57,7 +65,26 @@ func (k *keepers) Observation(ctx context.Context, rt types.ReportTimestamp, _ t
 	// should be more uniform for all nodes
 	keys := keyList(filterUpkeeps(results, ktypes.Eligible))
 
-	b, err := limitedLengthEncode(keys, maxObservationLength)
+	// Check limit
+	if len(keys) > observationUpkeepLimit {
+		keys = keys[:observationUpkeepLimit]
+	}
```

Comment:
nit. this might make more sense to do on `UpkeepIdentifiers` array after we've converted keys to upkeep identifiers